napari/utils/colormaps/colormap.py

```diff
@@ -44,10 +46,14 @@ class Colormap(EventedModel):
     # fields
     colors: Array[float, (-1, 4)]
     name: str = 'custom'
+    display_name: str = None
```

Comment:
Pulling out from other thread. Is it possible for us to make this be private? I think that would help keep the API clean. You might need to customize the init like we do here https://github.com/napari/napari/blob/25756da88a2ee09e8f6723994685e0aa6fbbb5af/napari/components/viewer_model.py#L111, and add **data, see https://pydantic-docs.helpmanual.io/usage/models/#private-model-attributes and elsewhere for details. It should be possible

Reply:
Thanks!

Reply:
could I ask to use `Optional` when `None` is a valid argument?

```python
display_name: Optional[str] = None
```

Reply:
@Czaki sure thing!

---

@sofroniewn just to check, the init of the class would still be `display_name` but the actual attribute would be private, correct ?

Reply:
yes i think so

Reply:
Fixed!